### PR TITLE
Feature/changeable image size

### PR DESCRIPTION
**Pull request by bzztbomb**
A couple of small things:

Compute the shape of the first FCL layer instead of using hardcoded numbers. Makes it easier to play with convolution layers and screen size.

Use batch_size variable in train.py

### CODE REVIEW

**Comment by bzier** [comment]
I'm new to TensorFlow and ML in general. Can you help me understand these sizes/shapes a little bit more? For instance, what did the hard-coded 1152 represent / where did it come from before? And now you are computing it instead, but I don't have an understanding as to how that computation is equivalent. If it's too much to explain & discuss in this PR, can you point me to any good resources to better understand these things? Thanks.

File: model.py [hunk 1]
```diff
@@ -49,11 +49,15 @@ def conv2d(x, W, stride):
 
 h_conv5 = tf.nn.relu(conv2d(h_conv4, W_conv5, 1) + b_conv5)
 
+# Poorly named probably, this converts the convolution layer into a flattened layer. Basically the shape of h_conv5 multiplied out
+s = h_conv5.get_shape().as_list()
+flatten_dimension = s[1] * s[2] * s[3]
```
Comment (kevinhughes27) [inline model.py]: how about `flattened_length` and then we remove the comment?

Comment (bzztbomb) [inline model.py]: Yeah, good call!

**Comment by bzztbomb** [comment]
@bzier I'm new to this stuff as well. I've gone through a bit of this online course:
https://www.kadenze.com/courses/creative-applications-of-deep-learning-with-tensorflow/info

Which has been helpful.  

The shape of a tensor is just the dimensions of the matrix.  So that's linear algebra stuff (https://ocw.mit.edu/courses/mathematics/18-06-linear-algebra-spring-2010/).  The 1152 was equal to the length of the last convolution matrix in the model.  (I think it was ?x1x18x64). It was easy to see using tensorboard, if you look at the autopilot repo its easy to add the tensorboard code back in so you can investigate the graph.

**Comment by bzier** [comment]
Thanks @bzztbomb. My linear algebra is a little rusty and Tensorflow is all new for me, so I'll definitely have to check out those courses. I appreciate it.

**Comment by kevinhughes27** [comment]
Yeah the sizes have to work out in a linear algebra way or else the layers can't be used in matrix operations. Some of the work is manual and some is calculated so this is a step in the right direction. If we switch to something like keras then a lot of this math goes away and the framework takes care of it.

**Pull request merged**